creator-node/test/processStateMonitoringJob.test.js

```diff
@@ -0,0 +1,427 @@
+    expect(getNodeUsersStub).to.have.been.calledOnceWithExactly(
+      DISCOVERY_NODE_ENDPOINT,
+      CONTENT_NODE_ENDPOINT,
+      LAST_PROCESSED_USER_ID,
+      NUM_USERS_TO_PROCESS
+    )
+    expect(getUnhealthyPeersStub).to.have.been.calledOnceWithExactly(USERS)
+    expect(
+      buildReplicaSetNodesToUserWalletsMapStub
+    ).to.have.been.calledOnceWithExactly(USERS)
+    expect(retrieveClockStatusesForUsersAcrossReplicaSetStub).to.not.have.been
+      .called
+    expect(computeUserSecondarySyncSuccessRatesMapStub).to.not.have.been.called
+    expect(aggregateReconfigAndPotentialSyncOpsStub).to.not.have.been.called
+    expect(jobResult).to.deep.equal({
+      lastProcessedUserId: USER_ID,
+      jobFailed: true
+    })
+  })
+
+  it('should return without throwing when getUnhealthyPeers throws an error', async function () {
+    // Run processStateMonitoringJob with each step succeeding except getUnhealthyPeers
+    getUnhealthyPeersStub = sandbox.stub().rejects('test unexpected error')
+    const jobResult = await processStateMonitoringJob({})
+
+    // Verify that each step was called with the expected params
+    expect(getNodeUsersStub).to.have.been.calledOnceWithExactly(
+      DISCOVERY_NODE_ENDPOINT,
+      CONTENT_NODE_ENDPOINT,
+      LAST_PROCESSED_USER_ID,
+      NUM_USERS_TO_PROCESS
+    )
+    expect(getUnhealthyPeersStub).to.have.been.calledOnceWithExactly(USERS)
+    expect(buildReplicaSetNodesToUserWalletsMapStub).to.not.have.been.called
+    expect(retrieveClockStatusesForUsersAcrossReplicaSetStub).to.not.have.been
+      .called
+    expect(computeUserSecondarySyncSuccessRatesMapStub).to.not.have.been.called
+    expect(aggregateReconfigAndPotentialSyncOpsStub).to.not.have.been.called
+    expect(jobResult).to.deep.equal({
+      lastProcessedUserId: USER_ID,
+      jobFailed: true
+    })
+  })
+
+  it('should return without throwing when getNodeUsers throws an error', async function () {
+    // Run processStateMonitoringJob with each step succeeding except getNodeUsers
+    getNodeUsersStub = sandbox.stub().rejects('test unexpected error')
+    const jobResult = await processStateMonitoringJob({})
+
+    // Verify that each step was called with the expected params
+    expect(getNodeUsersStub).to.have.been.calledOnceWithExactly(
+      DISCOVERY_NODE_ENDPOINT,
+      CONTENT_NODE_ENDPOINT,
+      LAST_PROCESSED_USER_ID,
+      NUM_USERS_TO_PROCESS
+    )
+    expect(getUnhealthyPeersStub).to.not.have.been.called
+    expect(buildReplicaSetNodesToUserWalletsMapStub).to.not.have.been.called
+    expect(retrieveClockStatusesForUsersAcrossReplicaSetStub).to.not.have.been
+      .called
+    expect(computeUserSecondarySyncSuccessRatesMapStub).to.not.have.been.called
+    expect(aggregateReconfigAndPotentialSyncOpsStub).to.not.have.been.called
+    expect(jobResult).to.deep.equal({
+      lastProcessedUserId: LAST_PROCESSED_USER_ID,
+      jobFailed: true
+    })
+  })
```

Comment:
these are all a variation of the same test: run `processStateMonitoringJob()` with one step failing and make sure the error gets caught and steps after it don't get run